src/Disks/IO/WriteBufferFromAzureBlobStorage.cpp

```diff
@@ -167,6 +169,25 @@ void WriteBufferFromAzureBlobStorage::finalizeImpl()
         execWithRetry([&](){ block_blob_client.CommitBlockList(block_ids); }, max_unexpected_write_error_retries);
         LOG_TRACE(log, "Committed {} blocks for blob `{}`", block_ids.size(), blob_path);
     }
+
+    if (check_objects_after_upload)
+    {
+        try
+        {
+            auto blob_client = blob_container_client->GetBlobClient(blob_path);
+            blob_client.GetProperties();
+            return;
+        }
+        catch (const Azure::Storage::StorageException & e)
+        {
+            if (e.StatusCode == Azure::Core::Http::HttpStatusCode::NotFound)
+                throw Exception(
+                        ErrorCodes::AZURE_BLOB_STORAGE_ERROR,
+                        "Object {} not uploaded to azure blob storage, it's a bug in Azure Blob Storage or its API.",
```

Comment:
> ... a bug in Azure Blob Storage or its API.

What is the history behind this PR? Did we encounter a bug at some point?

Reply:
discussed offline